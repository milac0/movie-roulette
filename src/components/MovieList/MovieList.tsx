import React, { useContext, useEffect } from "react";
import axios from "axios";
import { DataContext } from "../../context/DataContext";
import { Link } from "react-router-dom";
import MovieThumbnail from "../../components/MovieThumbnail/MovieThumbnail";
import styles from "./movieList.scss";

interface Props {}

const MovieList: React.FC<Props> = () => {
  const { filterBy, movies, setMovies, query, page } = useContext(DataContext);

  useEffect(() => {
    if (filterBy) {
      (async () => {
        const response = await axios.get(
          `/movie/${filterBy}?api_key=${process.env.API_KEY}&language=en-US&page=1`
        );
        setMovies(response.data.results);
      })();
    }
  }, [filterBy]);

  useEffect(() => {
    (async () => {
      const url = query
        ? `/search/movie?api_key=${process.env.API_KEY}&language=en-US&query=${query}&include_adult=false&page=${page}`
        : `/movie/${filterBy}?api_key=${process.env.API_KEY}&language=en-US&page=${page}`;
      const response = await axios.get(url);
      setMovies([...movies, ...response.data.results]);
    })();
  }, [page]);

  return (
    <div className={styles.list}>
      {movies.map((movie, i) => (
        <Link className={styles.link} to={`/movies/${movie.id}`} key={i}>
          <MovieThumbnail movie={movie} />
        </Link>
      ))}
    </div>
  );
};

export default MovieList;
