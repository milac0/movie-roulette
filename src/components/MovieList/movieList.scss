@import "../../theme/variables";

.list {
  display: grid;
  grid-template-columns: repeat(auto-fill, 220px);
  grid-gap: 0px;
  justify-content: center;
  margin: 1em auto;
  max-width: 1200px;
  width: 90%;
}

.link {
  color: $primary;
  text-decoration: none;
}
