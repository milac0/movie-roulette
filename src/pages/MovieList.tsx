import React, { useEffect, useState, Fragment, useContext } from "react";
import { css } from "emotion";
import axios from "axios";
import { Link } from "react-router-dom";
import MovieThumbnail from "../components/MovieThumbnail";
import { colors } from "./../theme/theme";
import RouletteButton from "../components/RouletteButton";
import { DataContext } from "./../context/DataContext";
import SearchBar from "../components/SearchBar";

const list = css`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: 1em auto;
  max-width: 1200px;
  width: 90%;
`;

const link = css`
  text-decoration: none;
  color: ${colors.primary};
`;

const buttons = css`
  text-align: center;
  margin-bottom: 2em;
  position: relative;

  button {
    background: ${colors.primary};
    border: none;
    border-radius: 5px;
    color: white;
    font-size: 1rem;
    margin: 0 auto;
    padding: 1em;
    width: 20em;

    &:hover {
      background: ${colors.primaryDark};
      cursor: pointer;
    }
  }
`;

interface Props {}

const MovieList: React.FC<Props> = () => {
  const { filterBy, movies, setMovies, query, page, setPage } = useContext(
    DataContext
  );

  useEffect(() => {
    if (filterBy) {
      (async () => {
        const response = await axios.get(
          `/movie/${filterBy}?api_key=${process.env.API_KEY}&language=en-US&page=1`
        );
        setMovies(response.data.results);
      })();
    }
  }, [filterBy]);

  useEffect(() => {
    (async () => {
      const url = query
        ? `/search/movie?api_key=${process.env.API_KEY}&language=en-US&query=${query}&include_adult=false&page=${page}`
        : `/movie/${filterBy}?api_key=${process.env.API_KEY}&language=en-US&page=${page}`;
      const response = await axios.get(url);
      setMovies([...movies, ...response.data.results]);
    })();
  }, [page]);

  return (
    <Fragment>
      <SearchBar />
      <div className={list}>
        {movies.map((movie, i) => (
          <Link className={link} to={`/movies/${movie.id}`} key={i}>
            <MovieThumbnail movie={movie} />
          </Link>
        ))}
      </div>
      <div className={buttons}>
        <button onClick={() => setPage(page + 1)}>load more...</button>
        <RouletteButton />
      </div>
    </Fragment>
  );
};

export default MovieList;
