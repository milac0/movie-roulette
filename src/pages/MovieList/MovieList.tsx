import React, { useEffect, Fragment, useContext } from "react";
import axios from "axios";
import { Link } from "react-router-dom";
import MovieThumbnail from "../../components/MovieThumbnail/MovieThumbnail";
import RouletteButton from "../../components/RouletteButton/RouletteButton";
import { DataContext } from "../../context/DataContext";
import SearchBar from "../../components/SearchBar/SearchBar";
import styles from "./movieList.scss";
import LoadMoreButton from "../../components/LoadMoreButton/LoadMoreButton";

interface Props {}

const MovieList: React.FC<Props> = () => {
  const { filterBy, movies, setMovies, query, page } = useContext(DataContext);

  useEffect(() => {
    if (filterBy) {
      (async () => {
        const response = await axios.get(
          `/movie/${filterBy}?api_key=${process.env.API_KEY}&language=en-US&page=1`
        );
        setMovies(response.data.results);
      })();
    }
  }, [filterBy]);

  useEffect(() => {
    (async () => {
      const url = query
        ? `/search/movie?api_key=${process.env.API_KEY}&language=en-US&query=${query}&include_adult=false&page=${page}`
        : `/movie/${filterBy}?api_key=${process.env.API_KEY}&language=en-US&page=${page}`;
      const response = await axios.get(url);
      setMovies([...movies, ...response.data.results]);
    })();
  }, [page]);

  return (
    <Fragment>
      <SearchBar />
      <div className={styles.list}>
        {movies.map((movie, i) => (
          <Link className={styles.link} to={`/movies/${movie.id}`} key={i}>
            <MovieThumbnail movie={movie} />
          </Link>
        ))}
      </div>
      <div className={styles.buttons}>
        <LoadMoreButton />
        <RouletteButton />
      </div>
    </Fragment>
  );
};

export default MovieList;
