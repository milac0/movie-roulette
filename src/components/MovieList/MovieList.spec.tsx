import React from 'react';
import renderer from 'react-test-renderer';

import MovieList from './MovieList';

describe('MovieList renders correctly', () => {
    test('snapshot renders', () => {
        const tree = renderer.create(<MovieList />).toJSON();
        expect(tree).toMatchSnapshot();
    });

});

