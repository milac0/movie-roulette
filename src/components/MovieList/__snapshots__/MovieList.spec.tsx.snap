// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList renders correctly snapshot renders 1`] = `
<div
  className="list"
/>
`;
